### DISCUSSION TITLE
Deploy request completed but still showing as Deploying

### DISCUSSION THREAD

**Discussion by jcohenho** [planetscale/discussion#192] [Bug Reports] [answered]
I just approved a DR which added an enum column to a table. While the Deploy Request says it has completed, it's still hanging on 'Deploying', and has been stuck like this for 30+ minutes and counting. I checked the migration changes on `main` and I can see the new enum column there. This is causing a couple different issues:

1. I am now prevented from deleting this branch and creating a new one due to its 'deploying' status. 
2. Usually when a non-null column is added, Planetscale will complain that the migration will fail if no direction is given on what default values to use to populate existing rows. In this case, I hadn't provided any default values, yet Planetscale automatically chose the first enum value to populate existing rows with. 

Update:
The Deploy Request finally completed after **1h 4m** :(

**Comment by fooforge** [comment]
@jcohenho Dang, that should not be happening. Can you drop us a quick note via https://planetscale.com/support with your account's email? We'll take a look at this then. Thank you!

**Comment by jgreet** [comment, marked as answer]
Thanks for opening a ticket. We'll have to look in to 1, but for 2 I believe that's the documented [ENUM behavior in MySQL](https://dev.mysql.com/doc/refman/8.0/en/enum.html).

> If an ENUM column is declared to permit NULL, the NULL value is a valid value for the column, and the default value is NULL. If an ENUM column is declared NOT NULL, its default value is the first element of the list of permitted values. 

Perhaps we should add a message to the deploy request calling this out though.

**Reply by jcohenho** [reply]
Interesting, thanks for pointing out the default ENUM behavior!

**Reply by orware** [reply]
And to close the loop on the deploy request completion time issue we did check in with our Engineering Team on it.

That situation did appear to be caused by a bug on our end which was active at the time you created your deploy request and a fix was merged in relatively quickly (which is also what helped to allow your deploy request to move into the completed state).

With the fix already in place that should not be an issue you run into again for subsequent deploy requests.

**Reply by jcohenho** [reply]
Thank you!
